File: app/src/main/java/com/example/hostelmanagementsystem/models/MessPlan.kt
package com.example.hostelmanagementsystem.models

data class MessPlan(
    val id: String,
    val planDate: String,
    val breakfast: String,
    val lunch: String,
    val eveningSnacks: String,
    val dinner: String,
    val notes: String,
    val createdAt: String,
    val updatedAt: String,
    val createdBy: User?
)

File: backend/models/MessPlan.js
const mongoose = require("mongoose");

const messPlanSchema = new mongoose.Schema(
  {
    planDate: {
      type: Date,
      required: true,
      unique: true
    },
    breakfast: {
      type: String,
      required: true,
      trim: true
    },
    lunch: {
      type: String,
      required: true,
      trim: true
    },
    eveningSnacks: {
      type: String,
      required: true,
      trim: true
    },
    dinner: {
      type: String,
      required: true,
      trim: true
    },
    notes: {
      type: String,
      trim: true,
      default: ""
    },
    createdBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true
    }
  },
  { timestamps: true }
);

module.exports = mongoose.model("MessPlan", messPlanSchema);
